src/Val.h

```diff
@@ -846,6 +846,12 @@ class TableVal : public Val, public notifier::Modifiable {
 	// takes ownership of the reference.
 	double CallExpireFunc(Val *idx);
 
+	// Enum for the different kinds of changes an &on_change handler can see
+	enum OnChangeType { element_new, element_changed, element_removed, element_expired };
```

Comment:
Thought we tend to use ALL_CAPS for enum value names?  Threw me off only a bit, but would still appeal to consistency.